### PR TITLE
fix: dcmaw-8877 encryption keys not re created if user has passcode but no biometrics

### PR DESCRIPTION
**Pull request by jdubey**
# DCMAW-8877 iOS | Encryption keys not re-created if user has passcode but no biometrics set up

Add ability to test whether device has a passcode setup but is not enrolled in biometrics.  Use this test to configure the access control of the SecureStorage appropriately.
Wrap keychain access in a Task on the main thread.  This is necessary because on some devices (like my iPad), without this change the passcode entry keypad appears "behind" the splash or unlock page, and will not display until the app is minimized, making it impossible to unlock.

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

~~- [ ] Met all accessibility requirements?~~
    ~~- [ ] Checked dynamic type sizes are applied~~
    ~~- [ ] Checked VoiceOver can navigate your new code~~
    ~~- [ ] Checked a user can navigate only using a keyboard around your new code~~

## Before merging your pull request:
- [ ] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

File: Sources/Application/SceneLifecycle.swift [hunk 1]
```diff
@@ -19,7 +19,7 @@ extension SceneLifecycle {
     
     func promptToUnlock() {
         coordinator?.evaluateRevisit {
-            windowManager?.hideUnlockWindow()
+            self.windowManager?.hideUnlockWindow()
```
Comment (DylanRJ) [inline Sources/Application/SceneLifecycle.swift:22]: Is this needed?

Comment (jdubey) [reply Sources/Application/SceneLifecycle.swift:22]: Yes, because this is now an escaping closure.  Compiler complains otherwise

Comment (DylanRJ) [reply Sources/Application/SceneLifecycle.swift:22]: Worth capturing self as unowned?
```swift
func promptToUnlock() {
    coordinator?.evaluateRevisit { [unowned self] in
        windowManager?.hideUnlockWindow()
    }
}
```
?

File: Sources/Application/MainCoordinator.swift [hunk 2]
```diff
@@ -44,21 +44,25 @@ final class MainCoordinator: NSObject,
...
-        if userStore.returningAuthenticatedUser {
-            do {
-                tokenHolder.accessToken = try userStore.secureStoreService.readItem(itemName: .accessToken)
-                homeCoordinator?.updateToken(accessToken: tokenHolder.accessToken)
-                action()
-            } catch {
-                print("Error getting token: \(error)")
+    func evaluateRevisit(action: @escaping () -> Void) {
+        Task {
+            await MainActor.run {
```
Comment (DylanRJ) [inline Sources/Application/MainCoordinator.swift:49]: Wouldn't marking this method (and the other dealing with local auth in the `LoginCoordinator`) as `@MainActor` achieve essentially the same thing without having to include a Task and having the closure be `@escaping`?
There was a pretty sizeable refactor to do away with using `waitForTruth` where it isn't absolutely necessary and avoid flakey tests so would certainly be on the side of that where possible.

Comment (jdubey) [reply Sources/Application/MainCoordinator.swift:49]: Still doesn't work.  I admit this is an edge case, as it only seems to happen on my iPad, so it is unlikely to be encountered by other users.  On the other hand, you are pretty much blocked from using the app with local auth if you run into it.

**Review by mcgarj — APPROVED** [review]
